sdgym/collect.py

```diff
@@ -16,20 +19,47 @@ def collect_results(input_path, output_file):
             If ``output_file`` is provided, the consolidated
             results will be written there. Otherwise, they
             will be written to ``input_path``/results.csv.
+        aws_key (str):
+            If an ``aws_key`` is provided, the given access
+            key id will be used to read from and/or write to
+            any s3 paths.
+        aws_secret (str):
+            If an ``aws_secret`` is provided, the given secret
+            access key will be used to read from and/or write to
+            any s3 paths.
     """
     print(f'Reading results from {input_path}')
     run_path = pathlib.Path(input_path)
 
     scores = []
-    for path in tqdm.tqdm(list(run_path.glob('**/*.csv'))):
-        scores.append(pd.read_csv(path))
+    if is_s3_path(input_path):
+        s3 = get_s3_client(aws_key, aws_secret)
+        bucket_name, key_prefix = parse_s3_path(input_path)
+        resp = s3.list_objects(Bucket=bucket_name, Prefix=key_prefix)
+        csv_files = [f for f in resp['Contents'] if f['Key'].endswith('.csv')]
+        for csv_file in csv_files:
+            obj = s3.get_object(Bucket=bucket_name, Key=csv_file['Key'])
+            scores.append(pd.read_csv(io.BytesIO(obj['Body'].read())))
+    else:
+        for path in tqdm.tqdm(list(run_path.glob('**/*.csv'))):
+            scores.append(pd.read_csv(path))
 
     scores = pd.concat(scores).drop_duplicates()
 
     if output_file:
         output = output_file
     else:
-        output = run_path / 'results.csv'
+        output = f'{input_path}/results.csv'
 
     print(f'Storing results at {output}')
-    scores.to_csv(output, index=False)
+    if is_s3_path(output):
```

Comment:
After addressing the comments form PR #77, this can be simplified to just calling `write_csv(scores...)` instead of `scores.to_csv`